test/extended/router/router.go

```diff
@@ -50,9 +50,17 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:operator.openshift.io
 	oc = exutil.NewCLIWithPodSecurityLevel("router-stress", admissionapi.LevelBaseline)
 
 	g.BeforeEach(func() {
-		var err error
-		host, err = exutil.WaitForRouterServiceIP(oc)
+		exists, err := exutil.DoesApiResourceExist(oc.AdminConfig(), "ingresscontrollers")
 		o.Expect(err).NotTo(o.HaveOccurred())
+		if exists {
+			host, err = exutil.WaitForRouterServiceIP(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			panic(err)
+		} else {
+			// microshift
```

Comment:
Wherever possible the condition should be free of any MicroShift awareness.

Reply:
ack, will remove as per https://github.com/openshift/origin/pull/27571/files#r1034996931